lib/manager/maven/extract.ts

```diff
@@ -310,18 +365,43 @@ export async function extractAllPackageFiles(
   packageFiles: string[]
 ): Promise<PackageFile[]> {
   const packages: PackageFile[] = [];
+  const additionalRegistryUrls = [];
+
   for (const packageFile of packageFiles) {
     const content = await readLocalFile(packageFile, 'utf8');
     if (content) {
-      const pkg = extractPackage(content, packageFile);
-      if (pkg) {
-        packages.push(pkg);
+      if (packageFile.endsWith('settings.xml')) {
+        const registries = extractRegistries(content);
+        if (registries) {
+          logger.debug(
+            { registries, packageFile },
+            'Found registryUrls in settings.xml'
+          );
+          additionalRegistryUrls.push(...registries);
+        }
       } else {
-        logger.debug({ packageFile }, 'can not read dependencies');
+        const pkg = extractPackage(content, packageFile);
+        if (pkg) {
+          packages.push(pkg);
+        } else {
+          logger.debug({ packageFile }, 'can not read dependencies');
+        }
       }
     } else {
       logger.debug({ packageFile }, 'packageFile has no content');
     }
   }
+  if (additionalRegistryUrls) {
+    for (const pkgFile of packages) {
+      for (const dep of pkgFile.deps) {
+        /* istanbul ignore else */
```

Comment:
Why we can't test the else? Will it never happen?

Reply:
It will never happen in real case scenarios, because the maven extractor will always add the default maven registry to the dependency here: https://github.com/renovatebot/renovate/blob/main/lib/manager/maven/extract.ts#L57

So this can only happen if we write a test especially for it or if the maven extraction changes.

If you have an idea on how to test it, then I can implement it.
However I am not that deep into jest mocking, that I am able to pull out a test for this is a reasonable time frame :)

Reply:
```suggestion
        /* istanbul ignore else: should never happen */
```

So we shoiuld at least have a comment here 😉